buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt

```diff
@@ -369,7 +366,12 @@ class PerformanceTestPlugin : Plugin<Project> {
     }
 
     private
-    fun Project.testResultsZipTaskFor(performanceTest: TaskProvider<PerformanceTest>, name: String): TaskProvider<Zip> =
+    fun Project.determineLocalPerformanceTestClass(): KClass<out PerformanceTest> {
+        return if (name == "buildScanPerformance") BuildScanPerformanceTest::class else PerformanceTest::class
```

Comment:
I think we should change the hierarchy, so we don't have to do this if here.
Let's only have `AbstractPerformanceTest`, `DistributedPerformanceTest` and `PerformanceTest`. `AbstractPerformance` would what is `ReportGenerationPerformanceTest` now, `DistributedPerformanceTest` and `PerformanceTest` being subclasses. Then `PerformanceTest` would have a flag to whether or not to create the report, default `false`. Then the `buildScanPerformance` subproject can activate that flag.

Reply:
Or we use composition instead of inheritance - we create a class `GenerateReportAction` and add it to `PerformanceTest` and `DistributedTest` with `doLast`, WDYT?

Reply:
When we already have task classes, I would prefer to not add the action in `doLast`, especially since the `doLast` action need to do slightly different things for both tests.

I also don't like adding `doLast` actions to custom task classes.

Feel free to try it out though, if the code is cleaner and better to understand then let's do it.